staging/src/k8s.io/client-go/tools/events/event_recorder.go

```diff
@@ -94,9 +98,23 @@ func (recorder *recorderImpl) makeEvent(refRegarding *v1.ObjectReference, refRel
 	if namespace == "" {
 		namespace = metav1.NamespaceDefault
 	}
+	// The referenced Object name may not be a valid name for Events
+	// and fail the creation because of this, so we replace it by
+	// an alphanumeric string.
+	// Ref: https://issues.k8s.io/127594
+	name := fmt.Sprintf("%s.%x", refRegarding.Name, t.UnixNano())
+	if len(apimachineryvalidation.NameIsDNSSubdomain(name, false)) > 0 {
+		suffix := fmt.Sprintf(".%x", t.UnixNano())
+		prefix := base64.RawURLEncoding.EncodeToString([]byte(refRegarding.Name))
+		if len(prefix)+len(suffix) > utilvalidation.DNS1123SubdomainMaxLength {
+			prefix = prefix[:utilvalidation.DNS1123SubdomainMaxLength-len(suffix)]
+		}
+		name = strings.Trim(prefix+suffix, "-_")
```

Comment:
I *think* (but double check me) that _ anywhere is considered invalid (not just beginning or end), so that may need removing or substituting, not just trimming from beginning or end

also, when we're trimming, it's then theoretically possible to have an empty string, so that case needs handling, I guess

Reply:
see https://github.com/kubernetes/kubernetes/pull/129790#discussion_r1931002836